### ISSUE TITLE
Bug: [prefer-nullish-coalescing] faulty detection of conditional tests with `ignoreConditionalTests: true`

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10153]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

[link](https://typescript-eslint.io/play/#ts=5.5.2&showAST=es&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3lAXPTMAligObwA%2B8KArhBANwBQoksCy6WARngcWZRp0mzcNDjwAZtRRgMhVFIAUADzwyA1iiQB3FAEp1KLbpQiA9OcRIYcORACe8OAAcbGNI0lKoFSl30LKzQACyRaYHguaixgJBA0FAByLFd3RkJJeCVvX3J-fX14AG8AXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6MAG1w2HAHdJ0JpAA0a9VnXZIleU3GcAwuKYATSvkp3pAFRT5UGfHESHsAF8AgF01YMCgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

### Repro Code

```TypeScript
declare const a: string | null;
declare const b: string | null;

declare function f(x: unknown): unknown;

// correctly reports
f(a || b);

// should but doesn't report
if (f(a || b)) {}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/prefer-nullish-coalescing": [
      "warn",
      {
        "ignoreConditionalTests": true
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expect that `if (f(a || b)) {}` should report, since `a || b` is being used as a function argument, not the test of an `if` statement.

### Actual Result

it doesn't report

### Additional Info

Just popping this up as a standalone issue for visibility/tracking. It will likely be addressed in https://github.com/typescript-eslint/typescript-eslint/pull/9924#discussion_r1795649396

cc @developer-bandi
